### PR TITLE
Adds db helper for swapping out foreign keys before create/update

### PR DESCRIPTION
**Pull request by cannikin**
Need this in order to support the new relationship stuff here https://github.com/redwoodjs/redwood/pull/412

### CODE REVIEW

**Comment by cannikin** [comment]
Let me know what you think of where I put this function @peterp. Since this file is already included in all services I figured it was the most DRY place to put it. In the actual service it looks like:

```js
import { db, foreignKeyReplacement } from 'src/lib/db'

export const createPost = ({ input }) => {
  return db.post.create({
    data: foreignKeyReplace(input),
  })
}
```

File: api/src/lib/db.js [hunk 1]
```diff
@@ -4,3 +4,28 @@
 import { PrismaClient } from '@prisma/client'
 
 export const db = new PrismaClient()
+
+// Using this as a workaround until Prisma has a chance to work on
+// https://github.com/prisma/prisma/issues/2152
+export const foreignKeyReplacement = (input) => {
+  let output = input
+  const foreignKeys = []
```
Comment (mohsen1) [inline api/src/lib/db.js:12]:
```suggestion
  const foreignKeys = Object.keys(input).filter(k => k.match(/Id$/))
```

File: api/src/lib/db.js [hunk 2]
```diff
@@ -4,3 +4,28 @@
...
+// https://github.com/prisma/prisma/issues/2152
+export const foreignKeyReplacement = (input) => {
+  let output = input
+  const foreignKeys = Object.keys(input).filter(k => k.match(/Id$/))
+
+  for (let key in input) {
+    if (key.match(/Id$/)) {
+      foreignKeys.push(key)
+    }
+  }
```
Comment (mohsen1) [inline api/src/lib/db.js]: You should remove these lines now
```suggestion
```

**Comment by cannikin** [comment]
Thanks for that suggestion @mohsen1, as I was writing that I figured there was a simpler way to get those keys!

**Pull request merged**